src/mlpack/core/data/load_csv.hpp

```diff
@@ -12,389 +42,277 @@
-        throw std::runtime_error(oss.str());
-      }
-
-      if (!canParse)
-      {
-        std::ostringstream oss;
-        oss << "LoadCSV::NonTransposeParse(): parsing error on line " << col
-            << "!";
-        throw std::runtime_error(oss.str());
-      }
-
-      ++row; col = 0;
+      std::ostringstream oss;
+      oss << "Cannot open file '" << filename << "'. " << std::endl;
+      throw std::runtime_error(oss.str());
     }
+
+    inFile.unsetf(std::ios::skipws);
   }
 
+  // Fucntions for Categorical Parse
+
   /**
-   * Parse a transposed matrix.
-   *
-   * @param inout Matrix to load into.
-   * @param infoSet DatasetMapper to load with.
-   */
+  * Parse a non-transposed matrix.
+  *
+  * @param inout Matrix to load into.
+  * @param infoSet DatasetMapper object to load with.
+  */
   template<typename T, typename PolicyType>
-  void TransposeParse(arma::Mat<T>& inout, DatasetMapper<PolicyType>& infoSet)
-  {
-    using namespace boost::spirit;
-
-    // Get matrix size.  This also initializes infoSet correctly.
-    size_t rows, cols;
-    GetTransposeMatrixSize<T>(rows, cols, infoSet);
-
-    // Set the matrix size.
-    inout.set_size(rows, cols);
-
-    // Initialize auxiliary variables.
-    size_t row = 0;
-    size_t col = 0;
-    std::string line;
-    inFile.clear();
-    inFile.seekg(0, std::ios::beg);
-
-    /**
-     * This is the parse rule for strings.  When we get a string we have to pass
-     * it to the DatasetMapper.
-     */
-    auto parseString = [&](iter_type const &iter)
-    {
-      // All parsed values must be mapped.
-      std::string str(iter.begin(), iter.end());
-      boost::trim(str);
-
-      inout(row, col) = infoSet.template MapString<T>(std::move(str), row);
-      ++row;
-    };
-
-    while (std::getline(inFile, line))
-    {
-      // Remove whitespace from either side.
-      boost::trim(line);
-
-      // Reset the row we are looking at.  (Remember this is transposed.)
-      row = 0;
-
-      // Now use boost::spirit to parse the characters of the line;
-      // parseString() will be called when a token is detected.
-      const bool canParse = qi::parse(line.begin(), line.end(),
-          stringRule[parseString] % delimiterRule);
-
-      // Make sure we got the right number of rows.
-      if (row != rows)
-      {
-        std::ostringstream oss;
-        oss << "LoadCSV::TransposeParse(): wrong number of dimensions (" << row
-            << ") on line " << col << "; should be " << rows << " dimensions.";
-        throw std::runtime_error(oss.str());
-      }
-
-      if (!canParse)
-      {
-        std::ostringstream oss;
-        oss << "LoadCSV::TransposeParse(): parsing error on line " << col
-            << "!";
-        throw std::runtime_error(oss.str());
-      }
-
-      // Increment the column index.
-      ++col;
-    }
-  }
+  void NonTransposeParse(arma::Mat<T>& inout,
+                         DatasetMapper<PolicyType>& infoSet);
```

Comment:
Why this function is private?

Reply:
They were already private from the start in the old parser, so I haven't changed them.